GramAddict/plugins/like_from_urls.py

```diff
@@ -38,42 +43,61 @@ def __init__(self):
             is_job_completed = False
 
         self.args = config.args
+        self.device = device
         self.device_id = config.args.device
         self.state = None
         self.sessions = sessions
         self.session_state = sessions[-1]
         self.current_mode = plugin
 
-        self.urls = []
-        if os.path.isfile(self.args.urls_file):
-            with open(self.args.urls_file, "r") as f:
-                self.urls = f.readlines()
+        file_list = [file for file in (self.args.interact_from_file)]
+        shuffle(file_list)
 
-        self.state = State()
         on_like = partial(
             _on_like, sessions=self.sessions, session_state=self.session_state
         )
+        for filename in file_list:
+            self.state = State()
 
-        @run_safely(
-            device=device,
-            device_id=self.device_id,
-            sessions=self.sessions,
-            session_state=self.session_state,
-        )
-        def job():
-            for url in self.urls:
-                url = url.strip().replace("\n", "")
-                if validate_url(url) and "instagram.com/p/" in url:
-                    if open_instagram_with_url(self.device_id, url) is True:
-                        opened_post_view = OpenedPostView(device)
-                        like_succeed = do_like(opened_post_view, device, on_like)
-                        logger.info(
-                            "Like for: {}, status: {}".format(url, like_succeed)
-                        )
+            @run_safely(
+                device=self.device,
+                device_id=self.device_id,
+                sessions=self.sessions,
+                session_state=self.session_state,
+            )
+            def job():
+                self.process_file(filename, on_like, storage)
 
-                        if like_succeed:
-                            logger.info("Back to profile")
-                            device.back()
-                            random_sleep()
+            job()
 
-        job()
+    def process_file(self, current_file, on_like, storage):
+        # TODO: We need to add interactions properly, honor session/source limits, honor filter,
```

Comment:
I think we shouldn't merge this to master until this feature respect limits and filter. Otherwise, we will introduce a feature with inconsistent behaviour compared to all other features.

Reply:
The issue is that this is already live. 😄 This is the _current_ behavior.

Reply:
If it's live already, then it's ok. But we need to fix that asap to avoid people getting blocked because they expected the limits to be respected as all other features.